src/python/grpcio/grpc/_common.py

```diff
@@ -96,3 +99,50 @@ def deserialize(serialized_message, deserializer):
 
 def fully_qualified_method(group, method):
     return '/{}/{}'.format(group, method)
+
+
+def _wait_once(wait_fn, timeout, spin_cb):
+    wait_fn(timeout=timeout)
+    if spin_cb is not None:
+        spin_cb()
+
+
+def wait(wait_fn, wait_complete_fn, timeout=None, spin_cb=None):
```

Comment:
Optional: if we move this function to Cython layer, we can gain free performance boost to reduce the overhead of introducing this spin wait mechanism.